catalog/spatial/wfs/spatial-wfs-mapper/src/test/java/org/codice/ddf/spatial/ogc/wfs/catalog/mapper/impl/TestMetacardMapperImpl.java

```diff
@@ -14,76 +14,245 @@
+            .get()
+            .getFeatureProperty(),
+        is(FEATURE_NAME));
+
+    assertThat(
+        metacardMapper
+            .getEntry(e -> e.getAttributeName().equals(METACARD_MODIFIED))
+            .get()
+            .getFeatureProperty(),
+        is(FEATURE_MODIFIED));
+
+    assertThat(
+        metacardMapper
+            .getEntry(e -> e.getAttributeName().equals(METACARD_KEYWORD_1))
+            .get()
+            .getFeatureProperty(),
+        is(FEATURE_CALLSIGN));
 
-    // Test
-    String featureProperty = metacardMapper.getFeatureProperty(Metacard.MODIFIED);
+    assertThat(
+        metacardMapper
+            .getEntry(e -> e.getAttributeName().equals(METACARD_KEYWORD_2))
+            .get()
+            .getFeatureProperty(),
+        is(FEATURE_MISSIONID));
 
-    // Verify
-    assertThat(featureProperty, is(nullValue()));
+    assertThat(
+        metacardMapper
+            .getEntry(e -> e.getAttributeName().equals(METACARD_CONTENTSTORE))
+            .get()
+            .getFeatureProperty(),
+        is(""));
   }
 
   @Test
-  public void testGetMetacardAttributeMappingGivenFeaturePropertyMappingExists() {
-    // setup
-    MetacardMapperImpl metacardMapper = new MetacardMapperImpl();
-    metacardMapper.setFeatureType(EXAMPLE_FEATURE_TYPE);
-    metacardMapper.setResourceSizeMapping(FEATURE_PROPERTY);
-    metacardMapper.setDataUnit("MB");
+  public void testGetMetacardAttribute() {
+    assertThat(
+        metacardMapper
+            .getEntry(e -> e.getFeatureProperty().equals(FEATURE_NAME))
+            .get()
+            .getAttributeName(),
+        is(METACARD_TITLE));
+
+    assertThat(
+        metacardMapper
+            .getEntry(e -> e.getFeatureProperty().equals(FEATURE_MODIFIED))
+            .get()
+            .getAttributeName(),
+        is(METACARD_MODIFIED));
 
-    // Test
-    String metacardAttribute = metacardMapper.getMetacardAttribute(FEATURE_PROPERTY);
-    String unit = metacardMapper.getDataUnit();
+    assertThat(
+        metacardMapper
+            .getEntry(e -> e.getFeatureProperty().equals(FEATURE_CALLSIGN))
+            .get()
+            .getAttributeName(),
+        is(METACARD_KEYWORD_1));
 
-    // Verify
-    assertThat(metacardAttribute, is(Metacard.RESOURCE_SIZE));
-    assertThat(unit, is("MB"));
+    assertThat(
+        metacardMapper
+            .getEntry(e -> e.getFeatureProperty().equals(FEATURE_MISSIONID))
+            .get()
+            .getAttributeName(),
+        is(METACARD_KEYWORD_2));
   }
 
   @Test
-  public void testGetMetacardAttributeMappingGivenFeaturePropertyMappingDoesNotExists() {
-    // setup
-    MetacardMapperImpl metacardMapper = new MetacardMapperImpl();
-    metacardMapper.setFeatureType(EXAMPLE_FEATURE_TYPE);
-    metacardMapper.setExpirationDateMapping(FEATURE_PROPERTY);
+  public void testGetEntry() {
+    Optional o = metacardMapper.getEntry(e -> e.getFeatureProperty().equals(FEATURE_NAME));
```

Comment:
nit: could make this variable name a little more descriptive

Reply:
expanded variable name